src/hotspot/os/linux/os_linux.cpp

```diff
@@ -6543,6 +6554,33 @@ bool CracRestoreParameters::read_from(int fd) {
   ::_restore_start_time = hdr->_restore_time;
   ::_restore_start_counter = hdr->_restore_counter;
 
+  for (int i = 0; i < hdr->_nflags; i++) {
```

Comment:
Makes sense, I'll turn those into errors. I should probably also check the presence of `-jar` and `-cp`/`--classpath` and produce a nice explanation; otherwise the code would interpret those as the new main class and its arguments.